crates/daemon/src/monitor.rs

```diff
@@ -1023,3 +1030,139 @@ static TRANSACTION_BROADCAST_COUNTER: conquer_once::Lazy<prometheus::IntCounterV
         )
         .unwrap()
     });
+
+fn batch_script_get_history(
+    client: Arc<electrum_client::Client>,
+    scripts: Vec<Script>,
+) -> tokio::sync::mpsc::Receiver<Vec<GetHistoryRes>> {
+    let (tx_script_updates, rx_script_updates) = tokio::sync::mpsc::channel(BATCH_SIZE * 4);
+
+    let batches = scripts.chunks(BATCH_SIZE).map(|batch| batch.to_owned());
+
+    // It's important to move here so the sender gets dropped and the receiver finishes correctly
```

Comment:
🙃 Given that you extracted this into a function, I think it should work with a `for` loop, since the original copy of the `sender` will be dropped as soon as you return from this function.

This is fine anyway, but the comment might not be entirely true anymore.